pkg/lib/registry/registry.go

```diff
@@ -342,9 +342,45 @@ func (r RegistryUpdater) DeprecateFromRegistry(request DeprecateFromRegistryRequ
 		return fmt.Errorf("unable to migrate database: %s", err)
 	}
 
-	deprecator := sqlite.NewSQLDeprecatorForBundles(dbLoader, request.Bundles)
+	// Check if all bundlepaths are valid
+	var toDeprecate []string
+
+	dbQuerier := sqlite.NewSQLLiteQuerierFromDb(db)
+	registryBundles, err := dbQuerier.ListBundles(context.TODO())
+	if err != nil {
+		if !request.Permissive {
+			return fmt.Errorf("cannot verify bundle presence in registry: %v", err)
+		}
+		toDeprecate = request.Bundles
```

Comment:
Is there any reason we wouldn't want to log that we ran into an error listing the bundles here, even at the debug level? I'm trying to wrap my head around the current permissive workflow, and whether we'd want to ignore if we weren't able to get the list of bundles in the database.

Reply:
It actually doesn't make much sense to try and remove bundles when you can't even locate them, permissive mode shouldn't even be used here. I changed to fail when those checks fail